interface_notes.md

```diff
@@ -0,0 +1,387 @@
+# What I'm doing
+As part of developing a CLI, I'm trying to standardize the interfaces. To do this, I'll impelement the following:
+- `interfaces.BaseInterface`: A base class for interfaces with appropriate functions
+  - This is complete other than `is_valid` which will need to call `is_valid` from a plugin's family.
+- `interfaces.BaseYamlInterface`: A base class for interfaces that make use of YAML configuration
+  - Not started
+- `interfaces.BaseFamily`: A base class for plugin families
+  - Not started
+  - Currently, this would implement a single method, `is_valid`.
+  - `is_valid` would require that sublcasses set `cls.required_args` and `cls.required_kwargs` as lists of arguments
+    that are required in the call signature.
+- `interfaces.BaseYamlFamily`: A base class for YAML plugin families
+  - Not Started
+  - Currently, this would implement a single method, `is_valid`.
+  - `is_valid` would validate yaml against some kind of schema that would be set in `cls.yaml_schema`.
+
+# Status
+- I have created a base interface class for all non-yaml-based plugins. The `BaseInterface` class implements several
+  functions:
+  - is_valid: Tests a specific plugin's call signature
+  - get: Gets a named plugin's callable
```

Comment:
@jsolbrig This is now "get_plugin" ?

Reply:
Yes, sorry, these are my notes that I intend to convert into documentation as part of #79. For now, though please consider this file to be just notes.